Client/Frontend/Home/TopSites/TopSitesDimension.swift

```diff
@@ -81,16 +80,16 @@ class TopSitesDimensionImplementation: TopSitesDimension {
     /// - Parameter interface: Tile number is based on layout, this param contains the parameters needed to computer the tile number
     /// - Returns: The number of tiles per row the user will see
     private func getNumberOfTilesPerRow(for interface: TopSitesUIInterface) -> Int {
-        if interface.isIphone {
-            return interface.isLandscape ? 8 : 4
+        let cellWidth = TopSitesViewModel.UX.cellEstimatedSize.width
+        let leadingInset = HomepageViewModel.UX.leadingInset(traitCollection: interface.trait,
+                                                             interfaceIdiom: interface.isIphone ? .phone : .pad)
```

Comment:
Instead of saving a boolean in the TopSitesUIInterface for isIphone we could save directly the userInterfaceIdiom, which would avoid us from doing this `isiPhone ?` check